src/test/ui/lint/no-coverage.rs

```diff
@@ -0,0 +1,46 @@
+#![feature(extern_types)]
+#![feature(no_coverage)]
+#![feature(type_alias_impl_trait)]
+#![warn(unused_attributes)]
+
+trait Trait {
```

Comment:
Consider adding `#[no_coverage]` to the trait and its impl too for additional test coverage.